sdk/core/azure-core/inc/azure/core/internal/io/file_helpers.hpp

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @file
+ * @brief Internal utility functions for files and directories.
+ *
+ */
+#pragma once
+
+#include <azure/core/platform.hpp>
+
+#include <stdexcept>
+#include <string>
+
+#include <sys/stat.h> // for stat() used to check file size and mkdir() used to create directories
+
+#if defined(AZ_PLATFORM_WINDOWS)
+#if !defined(WIN32_LEAN_AND_MEAN)
+#define WIN32_LEAN_AND_MEAN
+#endif
+#if !defined(NOMINMAX)
+#define NOMINMAX
+#endif
+#include <windows.h>
+#elif defined(AZ_PLATFORM_LINUX)
+#include <unistd.h>
+
+#include <sys/types.h>
+#endif
+#include <string.h>
+
+namespace Azure { namespace Core { namespace IO { namespace _internal {
+
+  /**
+   * @brief Add basic filesystem functionality.
+   */
+  class FileHelpers final {
```

Comment:
I might call this `FilesystemHelpers`, instead of `FileHelpers` because it's essentially returning filesystem related information about files, not information about the contents of the files themselves.

But I don't feel incredibly strongly about this.